src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -142,6 +153,10 @@ public function __construct(Collection|Database|Client|Manager|string $mongo, ar
         if ($this->initialTtl <= 0) {
             throw new InvalidTtlException(sprintf('"%s()" expects a strictly positive TTL, got "%d".', __METHOD__, $this->initialTtl));
         }
+
+        if (!empty($this->options['readPreference']) && !$this->options['readPreference'] instanceof ReadPreference) {
+            throw new InvalidArgumentException(sprintf('"%s()" expects a MongoDB\Driver\ReadPreference instance, got "%s".', __METHOD__, get_debug_type($this->options['readPreference'])));
+        }
```

Comment:
`isset` is more appropriate: if the value is an empty string or `0`, it is invalid.

```suggestion
        if (!isset($this->options['readPreference']) && !$this->options['readPreference'] instanceof ReadPreference) {
            throw new InvalidArgumentException(sprintf('"%s()" expects a MongoDB\Driver\ReadPreference instance, got "%s".', __METHOD__, get_debug_type($this->options['readPreference'])));
        }
```

Note that this the MongoDB driver, when the read preference is invalid, it's ignored and fallbacks to "primary".